server/tournament.py

```diff
@@ -544,6 +576,50 @@ def update_tournament_spirit_rankings(tournament: Tournament) -> None:
     tournament.save()
 
 
+def check_if_authorised_to_submit_scores_by_match(match: Match, player: Player) -> tuple[bool, int]:
```

Comment:
I think I prefer having `user` being passed here, rather than player. Just because we are technically checking for permissions of the "logged in user"...  For instance, we could have site admin users who are not players, at some point, who just do this kind of administration...

Reply:
hmm so for that we have direct access through the tournament manager page where we can control access by is_staff?

This function is especially to validate the logged in players

Reply:
Fair enough, for this function. 

But, when we say `can_<something>`, I feel the function should take a user, rather than a player...

Reply:
Hmm yep makes sense.. Will do the change and push

Reply:
So in MatchScore.. should the entered_by be User or Player? Currently ive kept as Player but what you think?